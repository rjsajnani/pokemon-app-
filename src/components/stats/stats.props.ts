export interface Stat {
  base_stat: number;
  stat: {
    name: string;
  };
}

export type StatsProps = {
  stats: Stat[];
};
