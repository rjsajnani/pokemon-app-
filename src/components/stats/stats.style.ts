import { StyleSheet } from 'react-native';

export const StatsStyle = StyleSheet.create({
  container: {
    padding: 20,
  },
  powerContainer: {
    flexDirection: 'row',
    paddingVertical: 5,
  },
  powerText: {
    width: '30%',
  },
  progressBarContainer: {
    width: '70%',
    flexDirection: 'row',
    alignItems: 'center',
  },
  number: {
    width: '12%',
    fontSize: 12,
    fontWeight: 'bold',
  },
  percentageBarContainer: {
    backgroundColor: '#dedede',
    width: 200,
    height: 5,
    borderRadius: 20,
    overflow: 'hidden',
  },
  bar: {
    height: 5,
    borderRadius: 20,
  },
});
