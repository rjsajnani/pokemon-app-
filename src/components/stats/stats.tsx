import { View, ViewStyle } from 'react-native';
import { Text } from '../common';
import { StatsStyle as styles } from './stats.style';
import { StatsProps as Props } from './stats.props';
import React from 'react';

export const Stats: React.FC<Props> = (props) => {
  const { stats } = props;

  const calculatePercentage = React.useMemo(
    () =>
      (value: number): ViewStyle => {
        return {
          backgroundColor: value > 49 ? '#00ac17' : '#ff3e3e',
          width: `${(value / 100) * 100}%`,
        };
      },
    [],
  );

  return (
    <View style={styles.container}>
      <Text text="Pokemon Stats" presets="header2" />
      {stats.map((stat, index) => {
        return (
          <View key={index} style={styles.powerContainer}>
            <Text
              presets="body2"
              text={stat.stat.name}
              style={styles.powerText}
            />
            <View style={styles.progressBarContainer}>
              <Text style={styles.number} text={stat.base_stat.toString()} />
              <View style={styles.percentageBarContainer}>
                <View
                  style={{
                    ...styles.bar,
                    ...calculatePercentage(stat.base_stat),
                  }}
                ></View>
              </View>
            </View>
          </View>
        );
      })}
    </View>
  );
};
