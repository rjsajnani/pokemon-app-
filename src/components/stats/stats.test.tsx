import React from 'react';
import { render, RenderAPI } from '@testing-library/react-native';

import { Stats } from './stats';
import { Stat } from './stats.props';

const dummyStats: Stat[] = [
  { base_stat: 20, stat: { name: 'HP' } },
  { base_stat: 50, stat: { name: 'Attack' } },
];

describe('<Stats />', () => {
  let instance: RenderAPI;
  beforeEach(function () {
    instance = render(<Stats stats={dummyStats} />);
  });

  it('renders correctly', () => {
    const tree = instance.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
