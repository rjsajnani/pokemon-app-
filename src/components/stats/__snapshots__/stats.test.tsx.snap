// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stats /> renders correctly 1`] = `
<View
  style={
    {
      "padding": 20,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "black",
          "fontSize": 20,
          "fontWeight": "600",
          "lineHeight": 29,
        },
        undefined,
      ]
    }
  >
    Pokemon Stats
  </Text>
  <View
    style={
      {
        "flexDirection": "row",
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "black",
            "fontSize": 14,
            "letterSpacing": 0.25,
            "lineHeight": 24,
          },
          {
            "width": "30%",
          },
        ]
      }
    >
      HP
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "width": "70%",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "black",
              "fontSize": 16,
              "letterSpacing": 0.2,
              "lineHeight": 24,
            },
            {
              "fontSize": 12,
              "fontWeight": "bold",
              "width": "12%",
            },
          ]
        }
      >
        20
      </Text>
      <View
        style={
          {
            "backgroundColor": "#dedede",
            "borderRadius": 20,
            "height": 5,
            "overflow": "hidden",
            "width": 200,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#ff3e3e",
              "borderRadius": 20,
              "height": 5,
              "width": "20%",
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "black",
            "fontSize": 14,
            "letterSpacing": 0.25,
            "lineHeight": 24,
          },
          {
            "width": "30%",
          },
        ]
      }
    >
      Attack
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "width": "70%",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "black",
              "fontSize": 16,
              "letterSpacing": 0.2,
              "lineHeight": 24,
            },
            {
              "fontSize": 12,
              "fontWeight": "bold",
              "width": "12%",
            },
          ]
        }
      >
        50
      </Text>
      <View
        style={
          {
            "backgroundColor": "#dedede",
            "borderRadius": 20,
            "height": 5,
            "overflow": "hidden",
            "width": 200,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#00ac17",
              "borderRadius": 20,
              "height": 5,
              "width": "50%",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
